protobuf/sync/v1/sync_service.proto

```diff
@@ -0,0 +1,49 @@
+syntax = "proto3";
+
+package sync.v1;
+
+option go_package = "flagd/grpcsync";
+
+/*
+   SyncFlagsRequest convey a key that identifies the flagd instance, enabling Sync provider to filter
+   specific flags for a specific flagd deployment.
+*/
+message SyncFlagsRequest {
+  string key = 1;
+}
+
+//SyncState convey the state of the payload
+enum SyncState {
+  // Value is ignored by the listening flagd
+  SYNC_STATE_UNSPECIFIED = 0;
+
+  // All the flags matching the request. This is the default response and other states can be ignored
+  // by the implementation. Flagd internally replace all existing flags for this response state.
+  SYNC_STATE_ALL = 1;
+
+  // Convey an addition of a flag. Flagd internally handles this by combining new flags with existing ones
+  SYNC_STATE_ADD = 2;
+
+  // Convey an update of a flag. Flagd internally try to update if the updated flag already exist OR if it is not,
+  // then it will get added
+  SYNC_STATE_UPDATE = 3;
+
+  // Convey a deletion of a flag. Flagd internally removes the flag
+  SYNC_STATE_DELETE = 4;
+
+  // Optional server ping to check client connectivity. Handling is ignored by flagd and is to merely support live check
+  SYNC_STATE_PING = 5;
+}
+
+// SyncFlagsResponse is the response contains flags and state
+message SyncFlagsResponse {
```

Comment:
should we include more context here, for example the source. In flags example it may be useful to know which feature flag configuration CR that we are reading from (assuming that the server will be serving a merged set of flags from multiple sources) for debugging

Reply:
hmm, I am not sure this is needed. From what I see, flagd should only care about sync provider and not its internals. We already store the flag "source" which we use for store maintenance operations (ex:- merging, adding ...) and this should be the correct boundary IMO. Sync provider grpc server implementation on the other hand is a different story but that's not relevant to flagd 🤔

Reply:
related: https://github.com/open-feature/schemas/pull/78#discussion_r1097841436